components/hook-query/node/pg.test.mjs

```diff
@@ -216,58 +203,90 @@ const proceedAsync = async () => {
   );
 };
 
-if (getOwnPropertyDescriptor(process.env, "TRAVIS")) {
-  proceedAsync();
-} else {
+// It would be nicer to use travis service:
+//
+// os: linux
+// dist: focal
+// service:
+//   - postgresql
+//
+// Unfortunatelly (wtf travis), it doesn't work:
+//
+// $ travis_setup_postgresql
+// Starting PostgreSQL v12
+// Assertion failed on job for postgresql@12-main.service.
+// sudo systemctl start postgresql@12-main
+//
+// Downgrading dist to bionic or xenial makes it work.
+// But then node 18 cannot be installed (wtf travis 2x).
+// The solution seems to restart the service with other conf.
+// At this point, it is not any better than to launch the service here.
+if (getPlatform() !== "win32") {
+  const { initdb, pg_isready, postgres } =
+    getPlatform() === "darwin"
+      ? { initdb: "initdb", pg_isready: "pg_isready", postgres: "postgres" }
+      : {
+          initdb: "/usr/lib/postgresql/13/bin/initdb",
+          pg_isready: "/usr/lib/postgresql/13/bin/pg_isready",
+          postgres: "/usr/lib/postgresql/13/bin/postgres",
+        };
+  const url = toAbsoluteUrl(getUuid(), getTmpUrl());
```

Comment:
I'm not sure how I feel about hardcoding this whole (travis-specific) test environment setup in the test code itself. 

Seems like the test should just expect pg to be set up correctly and up and running. Any necessary setup should be made in an external script.

Otherwise it's very difficult to run these tests locally.

Reply:
I agree with you. And I should do that for `mysql` as well. But this will be time consuming. I created an issue so that we can go ahead and merge this. https://github.com/getappmap/appmap-agent-js/pull/130